### PR TITLE
added gcp governance docs

### PR DESCRIPTION
**Pull request by joeyouss**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Added documentation for GCP governance 
* Jira/GitHub Issue numbers (if any): CCM-17310
* Preview links/images (Internal contributors only): 

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/2-Harness-Concepts.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+For example, a user can create an Enforcement to schedule the deletion of all GCP snapshots older than 14 days. This Enforcement will run on the **days specified by the user**, at the **specified time**, and with the **specified frequency (hourly, daily, monthly**). For instance, you could set it to run daily at 2:00 AM to ensure that any snapshots meeting the criteria are removed. Alternatively, you might choose to run it hourly during peak usage times, or monthly for less critical cleanup tasks. 
+
+While setting up a new Enforcement, you can select the following:
+- **Cloud provider**: Currently we support AWS, Azure and GCP.
+- **Rules/ Rule Sets**: You can select the Rules or Rule Sets that your Enforcement will consist of.
+- **Target Accounts and Regions**: The target accounts and regions that you will be running the Enforcements on.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/2-Harness-Concepts.md:124]: GCP doesn't have regions.

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/2-Harness-Concepts.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+After setting up the schedule, you can view the Enforcement on the **Enforcements** page. 
+
+<DocImage path={require('../static/enforcements_list_gcp.png')} width="90%" height="90%" title="Click to view full size image" />
+
+Furthermore, you can disable the Enforcement at any time using the toggle button in the **Status** column. If you want to turn off the dry-run mode, select **Edit** from the vertical ellipsis menu (⋮) then go to "Target And Schedule", use slider to turn off "Enforce Rule(s) in Dry Run mode" and click on Finish.
+
+### Update an Enforcement
+
+You can view any Enforcements on Rule Enforcements page. Click on the enforcement to view details such as the rules, target accounts, and regions included in the enforcement. For updating, you can use the "Edit" button from the vertical ellipsis menu (⋮) to update the enforcements as per your convenience.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/2-Harness-Concepts.md:165]: GCP doesn't have regions.

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
+---
+title: GCP Recommendations
```
Comment (iamanmolm) [inline docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md:2]: Kindly update this with the recommendation policies shared over slack.

**Review by iamanmolm — CHANGES_REQUESTED** [review]

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+description: This topic describes how to optimize cloud costs using asset governance.
+# sidebar_position: 2
+---
+
+Recommendations help kickstart your journey with governance. Essentially, Harness runs certain policies behind the scenes to generate recommendations for your governance-enabled Azure subscriptions. These policies not only help to cut costs but also increase the efficiency of your system. On the Governance Overview page, Harness showcases recommendations that will benefit you to save costs on associated resources. You can click on any recommendation to view its details.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md:7]: "governance-enabled GCP projects."

File: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+        value: 
+        op: ne
+    actions:
+      - type: snapshot
+        name_format: "{disk[name]:.50}-{now:%Y-%m-%d}"
+      - type: delete
+      
+```
+
+**Savings Computed:** Savings are determined as 35% of the total cost. Implementing this recommendation would result in a 35% reduction in the maintenance cost of that disk, bringing it down to 92%. Thus, we consider the minimum savings achievable, which is 35%.
```
Comment (iamanmolm) [inline docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md:239]: "Savings are considered as 35% of the total cost. Implementing this recommendation would result in 35% to 92% reduction in the maintenance cost of that disk. Thus, we considered the minimum savings achievable, which is 35%. Ref: https://cloud.google.com/compute/docs/viewing-and-applying-idle-resources-recommendations"

**Review by iamanmolm — CHANGES_REQUESTED** [review]

**Review by iamanmolm — DISMISSED** [review]

**Review by SmritiSatya — DISMISSED** [review]

**Pull request merged**